iOSClient/Activity/NCActivity.swift

```diff
@@ -351,89 +351,132 @@ extension NCActivity: UITableViewDataSource {
-                   partialResult.insert(newDay)
-            }.sorted(by: >)
-            self.tableView.reloadData()
-            NCUtility.shared.stopActivityIndicator()
-        }
+        newItems += activities.filter
+
+        self.allItems = newItems.sorted(by: { $0.dateKey > $1.dateKey })
+        self.sectionDates = self.allItems.reduce(into: Set<Date>()) { partialResult, next in
+            let newDay = Calendar.current.startOfDay(for: next.dateKey)
+            partialResult.insert(newDay)
+        }.sorted(by: >)
+        self.tableView.reloadData()
     }
     
-    @objc func loadActivity(idActivity: Int) {
-        
-        if !canFetchActivity { return }
-        canFetchActivity = false
-        
-        if idActivity > 0 {
-            
-            let height = self.tabBarController?.tabBar.frame.size.height ?? 0
-            NCUtility.shared.startActivityIndicator(backgroundView: self.view, blurEffect: false, bottom: height + 50, style: .gray)
+    func loadComments(disptachGroup: DispatchGroup? = nil) {
+        guard showComments, let metadata = metadata else { return }
+        disptachGroup?.enter()
+
+        NCCommunication.shared.getComments(fileId: metadata.fileId) { (account, comments, errorCode, errorDescription) in
+            if errorCode == 0, let comments = comments {
+                NCManageDatabase.shared.addComments(comments, account: metadata.account, objectId: metadata.fileId)
+            } else if errorCode != NCGlobal.shared.errorResourceNotFound {
+                NCContentPresenter.shared.messageNotification("_share_", description: errorDescription, delay: NCGlobal.shared.dismissAfterSecond, type: NCContentPresenter.messageType.error, errorCode: errorCode)
+            }
+
+            if let disptachGroup = disptachGroup {
+                disptachGroup.leave()
+            } else {
+                self.loadDataSource()
+            }
         }
-        
+    }
+
+    /// Check if most recent activivities are loaded, if not trigger reload
+    func checkRecentActivity(disptachGroup: DispatchGroup) {
+        let recentActivityId = NCManageDatabase.shared.getLatestActivityId(account: appDelegate.account)
+
+        guard recentActivityId > 0 else {
+            return self.loadActivity(idActivity: 0, disptachGroup: disptachGroup)
+        }
+
+        disptachGroup.enter()
+
+        NCCommunication.shared.getActivity(
+            since: 0,
+            limit: 1,
+            objectId: metadata?.fileId,
+            objectType: objectType,
+            previews: true) { (account, activities, errorCode, errorDescription) in
+                defer { disptachGroup.leave() }
+
+                guard errorCode == 0,
+                      account == self.appDelegate.account,
+                      let activity = activities.first,
+                      activity.idActivity > recentActivityId
+                else { return }
```

Comment:
Always return at the start (DB empty) check it

Reply:
Please before update the PR, do you the test.

Reply:
> Always return at the start (DB empty) check it

I'm not sure what you mean. If the DB is empty (aka `getLatestActivityId` == 0) I do return before requesting.. Feel free to elaborate..